hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/testing.md

```diff
@@ -288,6 +288,57 @@ For the default test dataset, hosted in the `landsat-pds` bucket, this is:
 </property>
 ```
 
+## <a name="versions"></a> Testing against versioned buckets
+
+AWS S3 and some third party stores support versioned buckets.
+
+Hadoop is adding awareness of this, including 
+
+* Using version ID to guarantee consistent reads of opened files.
+  [HADOOP-15625](https://issues.apache.org/jira/browse/HADOOP-15625)
+* Using version ID to guarantee consistent multipart copies.
+* Checks to avoid creating needless delete markers.
+
++ maybe more to come.
+
+To test these features, you need to have buckets with object versioning
+enabled.
+
+A full `hadoop-aws` test run implicitly cleans up all files in the bucket
+in `ITestS3AContractRootDir`, so all every test run creates a large set of
+old (deleted) file versions. To avoid large bills, you must
+create a lifecycle rule on the bucket to purge the old versions.
```

Comment:
maybe link to https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-lifecycle.html ?

Reply:
nevermind, given the specific steps listed below